stories/legend/9_color_picker.tsx

```diff
@@ -1,40 +1,45 @@
 import React, { useState } from 'react';
-import { number } from '@storybook/addon-knobs';
+import { action } from '@storybook/addon-actions';
 
-import { EuiColorPicker, EuiWrappingPopover } from '@elastic/eui';
+import { EuiColorPicker, EuiWrappingPopover, EuiButton, EuiSpacer } from '@elastic/eui';
 
-import { Axis, BarSeries, Chart, Position, ScaleType, Settings, PartialTheme, LegendColorPicker } from '../../src/';
-import { SeededDataGenerator } from '../../src/mocks/utils';
+import { Axis, BarSeries, Chart, Position, ScaleType, Settings, LegendColorPicker } from '../../src/';
+import { BARCHART_1Y1G } from '../../src/utils/data_samples/test_dataset';
+import { SeriesKey } from '../../src/chart_types/xy_chart/utils/series';
 
-const dg = new SeededDataGenerator();
+const onChangeAction = action('onChange');
+const onCloseAction = action('onClose');
 
 export const example = () => {
-  const theme: PartialTheme = {
-    legend: {
-      spacingBuffer: number('legend buffer value', 80),
-    },
-  };
-  const data = dg.generateGroupedSeries(5, 2);
-  const [colors, setColors] = useState<Record<string, string>>({});
+  const [colors, setColors] = useState<Record<SeriesKey, string>>({});
 
-  const renderColorPicker: LegendColorPicker = ({ anchor, color, onClose, seriesIdentifier }) => {
-    const handleChange = (color: string) => {
+  const renderColorPicker: LegendColorPicker = ({ anchor, color, onClose, seriesIdentifier, onChange }) => {
+    const handleClose = () => {
       onClose();
+      onCloseAction();
       setColors({
         ...colors,
         [seriesIdentifier.key]: color,
       });
     };
+    const handleChange = (color: string) => {
+      onChangeAction(color);
+      onChange(color);
```

Comment:
nit: just to keep consistency on the code between the handleChange and the handleClose
```suggestion
      onChange(color);
      onChangeAction(color);
```